mapeamento/mapeadores/spiders/mapeador_adiariosv1.py

```diff
@@ -0,0 +1,93 @@
+import datetime
+
+import scrapy
+
+from mapeadores.spiders.bases.mapeador_semantico import MapeadorSemantico
+from mapeadores.items import MapeamentoItem
+
+class MapeadorAdiarioV1(MapeadorSemantico):
+    name = "mapeador_adiariov1"
+
+    def pattern_name(self):
+        return "ADIARIOS_v1"
+
+    def urls_pattern(self, protocol, city, state_code):
+        # examples
+        # https://www.buriticupu.ma.gov.br/diariooficial.php
+        # https://www.anajatuba.ma.gov.br/diariooficial.php
+
+        return [
+            f"{protocol}://www.{city}.{state_code}.gov.br/diariooficial.php",
+        ]
+
+    def domain_generator(self, name):
+
+        cityname = self.remove_blankspaces(name)             # cityname
+        abbrev = self.name_abbreviation(name)                # cn 
+        
+        combination_list = [
+            cityname,
+            abbrev,
+        ]
+
+        name_parts = self.name_parts(name)                   # city | name
+        starts_with_prefeitura = self.add_prefeitura_to_names(combination_list) # pm/prefeitura + all
+        combination_list += name_parts + starts_with_prefeitura
+
+        return combination_list
```

Comment:
Na minha opinião seria mais fácil abordar o que `domain_generator` está fazendo de forma mais declarativa para não precisar reimplementar a função quase toda nas classes que herdam de `MapeadorSemantico`.

Se no final você quer uma lista apenas com as variações `['no_blankspace', 'abbreviated', 'name_parts', 'prepended_prefeitura']` essa lista poderia ser um atributo da classe e a `domain_generator` por padrão usa os outros métodos de acordo com o que você declarou.

Reply:
Neste caso, repensei a solução toda. Concordo que não tem que reimplementar a função nas classes herdeiras. 

Mas tb pensei que não seria bom ser declarativo para não correr o risco de não incluir uma variação por acidente ou desconhecimento. Uma pessoa contribuindo com uma nova classe filha não tem que conhecer todas as variações de antemão. 

E, como a lógica do mapeamento é a de ficar chutando mesmo, se uma variação for encontrada depois, é a classe MapeadorSemantico que tem que ser incrementada, pra possibilidade valer pra todos os mapeadores.